code/modules/mining/mine_turfs.dm

```diff
@@ -460,6 +460,20 @@ var/global/list/rockTurfEdgeCache
 	N.fullUpdateMineralOverlays()
 	return
 
+/turf/simulated/mineral/attack_hand(mob/living/user)
+	if(check_affinity(user.mind, AFFINITY_EARTH) && user.a_intent == "help")
```

Comment:
I'll split the witch check into a different proc so as not to convolute attack_hand().